packages/vue-router/src/createApp.ts

```diff
@@ -76,6 +76,12 @@ export const createApp = (
 
       history.replace(pathname);
 
+      // reload page on history change (back and forward)
+      window.onpopstate = () => {
```

Comment:
We are using a `history` abstraction on top of window history. We should not manipulate the window directly because it can create race conditions or strange behaviors. Also, window will break when doing server-side rendering. I think it's better we remove the performance optimization for now unless we can make it work without manipulating window history.